trade-finance-manager-ui/server/controllers/utilisation-reports/record-corrections/create-record-correction-request/index.ts

```diff
@@ -36,10 +45,63 @@ export const createRecordCorrectionRequest = async (req: CreateRecordCorrectionR
         facilityId: feeRecord.facilityId,
         exporter: feeRecord.exporter,
       },
+      formValues: {},
+      errors: EMPTY_CREATE_RECORD_CORRECTION_REQUEST_ERRORS_VIEW_MODEL,
+      backLinkHref: getLinkToPremiumPaymentsTab(reportId, [Number(feeRecordId)]),
+    });
+  } catch (error) {
+    console.error('Failed to get create record correction request', error);
+    return res.render('_partials/problem-with-service.njk', { user: req.session.user });
+  }
+};
+
+export type PostCreateRecordCorrectionRequestRequest = CustomExpressRequest<{
+  reqBody: CreateRecordCorrectionRequestFormRequestBody;
+  params: {
+    reportId: string;
+    feeRecordId: string;
+  };
+}>;
+
+/**
+ * Controller for the POST create record correction request route
+ * @param req - The request object
+ * @param res - The response object
+ */
+export const postCreateRecordCorrectionRequest = async (req: PostCreateRecordCorrectionRequestRequest, res: Response) => {
+  try {
+    const { user, userToken } = asUserSession(req.session);
+    const { reportId, feeRecordId } = req.params;
+
+    const formValues = extractCreateRecordCorrectionRequestFormValues(req.body);
+
+    const errors = validateCreateRecordCorrectionRequestFormValues(formValues);
+    const formHasErrors = errors.errorSummary.length !== 0;
+
+    if (!formHasErrors) {
+      return res.redirect(`/utilisation-reports/${reportId}/create-record-correction-request/${feeRecordId}/check-the-information`);
+    }
+
+    const feeRecord = await api.getFeeRecord(reportId, feeRecordId, userToken);
```

Comment:
Is it safe to assume the API returns `facilityId` and `exporter`?

Reply:
These are both typed as `string` on the response body so should both be defined, if this is not the case for some reason the catch block should catch this type error and render the "problem with service" page